### DISCUSSION TITLE
Admin's "Move Category" allows a category to be moved to one with products?

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#6166] [Ideas]
zc200.  Site with demo products.

- Navigate to **_Catalog :: Categories/Products_**.
- Click the "Move" (M) button associated with the "Gift Certificates" category.
- When presented with the dropdown list of categories, choose "Music" (a category with products) and click the "Move" button.
- The "Gift Certificates" category now exists as a sub-category of the "Music" category ... along with the music product.
- The following message is presented: 
![image](https://github.com/zencart/zencart/assets/2685585/1e0cae32-30bc-4277-8cb9-004bbee3aa98)

Shouldn't that move be disallowed?  That's going to wreck havoc on the storefront.

**Comment by torvista** [comment]
I've allowed creating a subcategory in a category-with-products, with a javascript alert/pre-warning to the same effect: "Are you sure? If you can't finish this in a few minutes, do it later when you can, Proceed/Cancel".
It's useful to relocate products into a new subcategory....but that subcategory needs creating first. So if the category creation  was prevented, the process becomes much more convoluted.
Not quite the same as this, but still, I think it should be allowed: there **is** a warning and one assumes the Admin is a grown-up who can read.

**Comment by scottcwilson** [comment]
We've discussed this in the past and the thinking was that it was useful as a transitional configuration, to be resolved with subsequent moves, and that disallowing it would make making some kinds of change very difficult.

**Comment by scottcwilson** [comment]
Related: https://github.com/zencart/zencart/pull/3334

**Comment by drbyte** [comment]
Would it make sense as a safeguard to set the category to **disabled** if it's just been moved into a category containing products? At least that would, in the short term, prevent catalog-side chaos. The admin could override it of course, but if they don't know what they've just done and freeze up, then at least the disabled status protects things until they move it someplace else or relocate products to someplace else.

I'd posted some other suggestions later on in the 3334 PR as well.

**Comment by scottcwilson** [comment]
Suggest closing in the spirit that #3334 was closed.  Who knows if this capability is required temporarily during a transition.

**Comment by scottcwilson** [comment]
Moving to discussion.
